kafka/flink-capitalizer/src/main/java/com/example/DataStreamJob.java

```diff
@@ -0,0 +1,82 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import org.apache.flink.api.common.eventtime.WatermarkStrategy;
+import org.apache.flink.api.common.serialization.SimpleStringSchema;
+import org.apache.flink.connector.base.DeliveryGuarantee;
+import org.apache.flink.connector.kafka.sink.KafkaRecordSerializationSchema;
+import org.apache.flink.connector.kafka.sink.KafkaSink;
+import org.apache.flink.connector.kafka.source.KafkaSource;
+import org.apache.flink.connector.kafka.source.enumerator.initializer.OffsetsInitializer;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+
+import java.util.Properties;
+
+/**
+ * An example Apache Flink® job that connects to
+ * <a href="https://developer.aiven.io/docs/products/kafka/howto/flink-with-aiven-for-kafka">Aiven
+ * for Apache Kafka®</a>.
+ */
+public class DataStreamJob {
+
+
+    public static void main(String[] args) throws Exception {
+        // Sets up the execution environment, which is the main entry point
+        // to building Flink applications.
+        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+        Properties props = new Properties();
+        props.put("security.protocol", "SSL");
+        props.put("ssl.endpoint.identification.algorithm", "");
+        props.put("ssl.truststore.location", "TRUSTSTORE_PATH/client.truststore.jks");
+        props.put("ssl.truststore.password", "KEY_TRUST_SECRET");
+        props.put("ssl.keystore.type", "PKCS12");
+        props.put("ssl.keystore.location", "KEYSTORE_PATH/client.keystore.p12");
+        props.put("ssl.keystore.password", "KEY_TRUST_SECRET");
+        props.put("ssl.key.password", "KEY_TRUST_SECRET");
+
+        KafkaSource<String> source = KafkaSource.<String>builder()
+                .setBootstrapServers("APACHE_KAFKA_HOST:APACHE_KAFKA_PORT")
+                .setGroupId("test-flink-input-group")
+                .setTopics("test-flink-input")
+                .setProperties(props)
+                .setStartingOffsets(OffsetsInitializer.earliest())
+                .setValueOnlyDeserializer(new SimpleStringSchema())
+                .build();
+
+        KafkaSink<String> sink = KafkaSink.<String>builder()
+                .setBootstrapServers("APACHE_KAFKA_HOST:APACHE_KAFKA_PORT")
```

Comment:
Maybe the placeholders could be extracted to static members?
```suggestion
public class DataStreamJob {

    private static final String APACHE_KAFKA_HOST = "REPLACE_WITH_KAFKA_HOST";
    private static final int APACHE_KAFKA_PORT = 9092;

    private static final String KEYSTORE_PATH = "REPLACE_WITH_PATH_TO_KEYSTORE";
    private static final String TRUSTSTORE_PATH = "REPLACE_WITH_PATH_TO_TRUSTSTORE";
    private static final String KEY_TRUST_SECRET = "REPLACE_WITH_KEY_TO_TRUST_SECRET";

    public static void main(String[] args) throws Exception {
        // Sets up the execution environment, which is the main entry point
        // to building Flink applications.
        final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        Properties props = new Properties();
        props.put("security.protocol", "SSL");
        props.put("ssl.endpoint.identification.algorithm", "");
        props.put("ssl.truststore.location", String.format("%s/client.truststore.jks", TRUSTSTORE_PATH));
        props.put("ssl.truststore.password", KEY_TRUST_SECRET);
        props.put("ssl.keystore.type", "PKCS12");
        props.put("ssl.keystore.location", String.format("%s/client.keystore.p12", KEYSTORE_PATH));
        props.put("ssl.keystore.password", KEY_TRUST_SECRET);
        props.put("ssl.key.password", KEY_TRUST_SECRET);

        String bootstrapServer = String.format("%s:%d", APACHE_KAFKA_HOST, APACHE_KAFKA_PORT);
        KafkaSource<String> source = KafkaSource.<String>builder()
                .setBootstrapServers(bootstrapServer)
                .setGroupId("test-flink-input-group")
                .setTopics("test-flink-input")
                .setProperties(props)
                .setStartingOffsets(OffsetsInitializer.earliest())
                .setValueOnlyDeserializer(new SimpleStringSchema())
                .build();

        KafkaSink<String> sink = KafkaSink.<String>builder()
                .setBootstrapServers(bootstrapServer)
```

Reply:
I believe we usually have just placeholders, we shouldn't overcook this :)

Reply:
Not a blocker for me.

Reply:
There was some discussion about redoing all of the code examples for articles with environment variables instead of placeholders :heart: .  If we go forward with that, I'm committing to fixing it here and in the article!

Reply:
agreed!